input/Glfw/GlfwInputDevice.cpp

```diff
@@ -351,6 +367,139 @@ namespace NovelRT::Input::Glfw
+
+        for (auto& mapped : _mappedActions)
+        {
+            if (mapped.pairedKey.GetExternalKeyCode() != key)
+                continue;
+
+            KeyStateFrameChangeLog log{};
+            bool mouseMod = false;
+            for (auto& [currentKey, currentLog] : map)
+            {
+                if (currentKey != mapped.pairedKey.GetExternalKeyCode())
+                    continue;
+
+                log = currentLog;
+                if (currentKey == GLFW_MOUSE_BUTTON_LEFT || currentKey == GLFW_MOUSE_BUTTON_MIDDLE ||
+                    currentKey == GLFW_MOUSE_BUTTON_RIGHT)
+                {
+                    mouseMod = true;
+                }
+                break;
+            }
+
+            if (mouseMod)
+            {
+                auto currentState = log.GetCurrentState();
+                if (state == (int32_t)KeyState::KeyDown &&
+                    (currentState == KeyState::KeyDown || currentState == KeyState::KeyDownHeld))
+                {
+                    state = (int32_t)KeyState::KeyDownHeld;
+                }
+            }
+
+            log.PushNewState(static_cast<KeyState>(state));
+            map.insert_or_assign(key, log);
+        }
+    }
+
+    void GlfwInputDevice::ProcessCursorMovement(NovelRT::Maths::GeoVector2F& pos)
+    {
+        _mousePos = DetermineMouseScreenPosition(pos);
+    }
+
+    void GlfwInputDevice::ProcessKeyState(int32_t key, KeyState state)
+    {
+        auto& previousBuffer = _keyStates.at(_previousBufferIndex);
+        auto& currentBuffer = _keyStates.at(_currentBufferIndex);
+
+        KeyState previousStateResult = KeyState::Idle;
+
+        for (auto& [previousKey, previousLog] : previousBuffer)
+        {
+            if (previousKey == key)
+            {
+                previousStateResult = previousLog.GetCurrentState();
+                break;
+            }
+        }
+
+        KeyStateFrameChangeLog changeLogObject = KeyStateFrameChangeLog();
+
+        for (auto& [currentKey, currentLog] : currentBuffer)
+        {
+            if (currentKey == key)
+            {
+                changeLogObject = currentLog;
+                break;
+            }
+        }
+
+        switch (state)
+        {
+            case KeyState::KeyDown:
+                if (previousStateResult == KeyState::KeyDown)
+                {
+                    changeLogObject.PushNewState(KeyState::KeyDownHeld);
+                }
+                else if (previousStateResult != KeyState::KeyDownHeld)
+                {
+                    changeLogObject.PushNewState(KeyState::KeyDown);
+                }
+                break;
+            case KeyState::KeyDownHeld:
+            case KeyState::KeyUp:
+                changeLogObject.PushNewState((previousStateResult == KeyState::KeyUp) ? KeyState::Idle : state);
+                break;
+            case KeyState::Idle:
+            default:
+                break;
```

Comment:
I'm guessing we just forgot to add braces to some of the bodies here.

Reply:
Added braces